framework/include/actioncomponents/ActionComponent.h

```diff
@@ -78,12 +84,15 @@ class ActionComponent : public Action
   void checkRequiredTasks() const;
 
   /// Get problem from action warehouse
-  FEProblem & getProblem()
+  FEProblemBase & getProblem()
   {
-    mooseAssert(_awh.problem().get(), "There should be a problem");
-    return *_awh.problem().get();
+    mooseAssert(_awh.problemBase().get(), "There should be a problem");
+    return *_awh.problemBase().get();
   }
 
+  /// Get the factory to build (often physics-related but not always) objects (for example a Positions)
+  virtual Factory & getFactory() const { return _factory; }
```

Comment:
There are derived objects that would override this?

Reply:
will remove virtual. this was copied from PhysicsBase. I ll fix it there